Corona-Warn-App/src/test/java/de/rki/coronawarnapp/tracing/ui/details/TracingDetailsItemProviderTest.kt

```diff
@@ -0,0 +1,349 @@
+import io.mockk.MockKAnnotations
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runBlockingTest
+import org.joda.time.Instant
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import testhelpers.BaseTest
+import testhelpers.coroutines.test
+
+class TracingDetailsItemProviderTest : BaseTest() {
+
+    @MockK(relaxed = true) lateinit var context: Context
+    @MockK(relaxed = true) lateinit var resources: Resources
+    @MockK(relaxed = true) lateinit var aggregatedRiskResult: AggregatedRiskResult
+
+    @MockK lateinit var tracingStatus: GeneralTracingStatus
+    @MockK lateinit var tracingRepository: TracingRepository
+    @MockK lateinit var riskLevelStorage: RiskLevelStorage
+
+    @BeforeEach
+    fun setup() {
+        MockKAnnotations.init(this)
+        every { context.resources } returns resources
+    }
+
+    @AfterEach
+    fun teardown() {
+        clearAllMocks()
+    }
+
+    private fun createInstance() = TracingDetailsItemProvider(
+        tracingStatus = tracingStatus,
+        tracingRepository = tracingRepository,
+        riskLevelStorage = riskLevelStorage
+    )
+
+    private fun prepare(
+        status: GeneralTracingStatus.Status,
+        activeTracingDaysInRetentionPeriod: Long,
+        riskLevel: ProtoRiskLevel,
+        matchedKeyCount: Int
+    ) {
+        every { tracingStatus.generalStatus } returns flowOf(status)
+        every { tracingRepository.activeTracingDaysInRetentionPeriod } returns flowOf(activeTracingDaysInRetentionPeriod)
+        every { aggregatedRiskResult.totalRiskLevel } returns riskLevel
+
+        if (riskLevel == ProtoRiskLevel.LOW) {
+            every { aggregatedRiskResult.isLowRisk() } returns true
+        } else if (riskLevel == ProtoRiskLevel.HIGH) {
+            every { aggregatedRiskResult.isIncreasedRisk() } returns true
+        }
+
+        val exposureWindow: ExposureWindow = mockk()
+
+        val riskLevelResult = RiskLevelTaskResult(
+            calculatedAt = Instant.EPOCH,
+            aggregatedRiskResult = aggregatedRiskResult,
+            exposureWindows = listOf(exposureWindow)
+        )
+        every { riskLevelResult.matchedKeyCount } returns matchedKeyCount
+        every { riskLevelStorage.latestAndLastSuccessful } returns flowOf(listOf(riskLevelResult))
+    }
+
+    @Test
+    fun `additional info low risk box`() = runBlockingTest {
+
+        prepare(
+            status = GeneralTracingStatus.Status.TRACING_ACTIVE,
+            activeTracingDaysInRetentionPeriod = 0,
+            riskLevel = ProtoRiskLevel.LOW,
+            matchedKeyCount = 1
+        )
+
+        val instance = createInstance()
+        val testCollector = instance.state.test(startOnScope = this)
+
+        testCollector.latestValue?.size?.shouldBeGreaterThan(0)
```

Comment:
Do not use `?.` in test assertions, it will silently fail without anyone knowing.

There needs to be a latest value, but if there is none, this will still return true.